Add API key guide and info in API Overview

hillaryfraley opened the pull request:
## Description
Adds API keys info from issue https://github.com/sensu/sensu-docs/issues/1834

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/1834

## Review Instructions
I created a page in the Guides section and an identical section in the API Overview
@cwjohnston I would like to get your opinion on how this is documented -- I'm not sure it should stay the way it is. Seems thinner than our other guides.

Related PRs of note:
- https://github.com/sensu/sensu-docs/pull/1799
- https://github.com/sensu/sensu-docs/pull/1800

hillaryfraley committed:
add apikeys guide and info in api overview

cwjohnston commented on content/sensu-go/5.15/api/overview.md:
```diff
@@ -135,6 +135,93 @@ http://127.0.0.1:8080/api/core/v2/namespaces/default/events
+#### Example
+
+{{< highlight shell >}}
+$ curl -H "Authorization: Key 7f63b5bc-41f4-4b3e-b59b-5431afd7e6a2" 
```
This curl command appears incomplete, probably needs a URL added, e.g. `http://127.0.0.1:8080/api/core/v2/namespaces/default/checks`

hillaryfraley replied:
```suggestion
$ curl -H "Authorization: Key 7f63b5bc-41f4-4b3e-b59b-5431afd7e6a2" http://127.0.0.1:8080/api/core/v2/namespaces/default/checks
```

cwjohnston commented on content/sensu-go/5.15/api/overview.md:
```diff
@@ -135,6 +135,93 @@ http://127.0.0.1:8080/api/core/v2/namespaces/default/events
 
+### Authenticate with the API key feature
+
+The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username. API keys are cluster-wide resources, so only cluster admins can grant, view, and revoke them.
```
Present both here and in the guide below, I believe this description needs to be fleshed out some. Specifically, I think the description should answer the question "Why would I want to use API keys instead of access tokens described above?"

From my perspective there are multiple advantages:
* Check plugins, handler plugins and other code can integrate with the Sensu API without implementing the logic required to authenticate via the `/auth` API endpoint and periodically refresh the access token
* Username + password doesn't need to be provided in check or handler definitions (this is a big deal for a lot of security-conscious teams)
* API keys can be created/revoked without changing the underlying user's password
* Conversely (I expect that) API keys continue to work even if the user's password is changed.

cwjohnston commented on content/sensu-go/5.15/guides/use-apikey-feature.md:
```diff
@@ -0,0 +1,123 @@
+### Example
+
+{{< highlight shell >}}
+$ curl -H "Authorization: Key 7f63b5bc-41f4-4b3e-b59b-5431afd7e6a2" 
```
This command appears incomplete. See similar comment above.

hillaryfraley replied:
```suggestion
$ curl -H "Authorization: Key 7f63b5bc-41f4-4b3e-b59b-5431afd7e6a2" http://127.0.0.1:8080/api/core/v2/namespaces/default/checks
```

hillaryfraley commented:
Should we add something about API keys in https://docs.sensu.io/sensu-go/5.14/installation/auth/? Maybe not -- I don't see the token mentioned there.

hillaryfraley commented on content/sensu-go/5.15/api/overview.md:
```diff
@@ -135,6 +135,93 @@ http://127.0.0.1:8080/api/core/v2/namespaces/default/events
 
+### Authenticate with the API key feature
+
+The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username. API keys are cluster-wide resources, so only cluster admins can grant, view, and revoke them.
```
```suggestion
The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username. The advantages of authenticating with API keys rather than [access tokens](#authentication-quick-start) include:
- **More efficient integration**: Check and handler plugins and other code can integrate with the Sensu API without implementing the logic required to authenticate via the `/auth` API endpoint to periodically refresh the access token.
- **Improved security**: API keys do not require providing a username and password in check or handler definitions.
- **Better admin control**: API keys can be created and revoked without changing the underlying user's password...but keep in mind that API keys will continue to work even if the user's password changes.

API keys are cluster-wide resources, so only cluster admins can grant, view, and revoke them.
```

hillaryfraley replied:
What do you think of ^?

cwjohnston replied:
Let's copy these comments to the guide and :shipit:

hillaryfraley replied:
Done! Thank you.

hillaryfraley committed:
Update content/sensu-go/5.15/api/overview.md

hillaryfraley committed:
Update content/sensu-go/5.15/guides/use-apikey-feature.md

hillaryfraley committed:
Merge branch 'master' into doc-api-keys

hillaryfraley committed:
Remove Sensuctl mgmt commands for API keys

Cameron noted this information is more appropriate for the guide than the overview.

Commit:
Apply suggestions from code review

Co-Authored-By: Hillary Fraley <19190208+hillaryfraley@users.noreply.github.com>

Commit:
Merge branch 'master' into doc-api-keys

hillaryfraley commented on content/sensu-go/5.15/guides/use-apikey-feature.md:
```diff
@@ -0,0 +1,123 @@
+- [API key authentication](#api-key-authentication)
+- [Sensuctl management commands](#sensuctl-management-commands)
+
+The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username. API keys are cluster-wide resources, so only cluster admins can grant, view, and revoke them.
```
```suggestion
The Sensu API key feature (core/v2.APIKey) is a persistent UUID that maps to a stored Sensu username. The advantages of authenticating with API keys rather than [access tokens](#authentication-quick-start) include:
- **More efficient integration**: Check and handler plugins and other code can integrate with the Sensu API without implementing the logic required to authenticate via the `/auth` API endpoint to periodically refresh the access token.
- **Improved security**: API keys do not require providing a username and password in check or handler definitions.
- **Better admin control**: API keys can be created and revoked without changing the underlying user's password...but keep in mind that API keys will continue to work even if the user's password changes.

API keys are cluster-wide resources, so only cluster admins can grant, view, and revoke them.
```

hillaryfraley committed:
Update content/sensu-go/5.15/guides/use-apikey-feature.md
